chapters/2-collaboration.tex

```diff
@@ -217,16 +225,22 @@ \subsection{Choosing software}
 While this means you will inevitably have different projects
 with different code environments, each one will be better than the last,
 and you will avoid the costly process of migrating an ongoing project
-into a new code enviroment.
-Code environment should be constant down to the software level:
-the specific versions of software and the individual packages you use
+into a new code environment.
+Code environments should be documented as precise as possible.
+The specific version number of the programming languages and the individual packages you use
 should be referenced or maintained so that they can be reproduced going forward,
 even if different releases contain changes that would break your code
 or change your results.
   \index{software versions}
 DIME Analytics developed the command \texttt{ieboilstart} in the \texttt{ietoolkit} package
-to support Stata version stability.\sidenote{
+to support version and settings stability in Stata.\sidenote{
   \url{https://dimewiki.worldbank.org/ieboilstart}}
+If your project requires more than one programming language,
+for example if you analyze your data in one language but visualize your results in another,
+then make sure to make an as clear division between the two as possible. 
+This means that you first complete all tasks done in one language, 
+before the completing the rest of the tasks in the other language.
+Frequently swapping back and forth between languages is a reproducibility nightmare.
```

Comment:
```suggestion
If your project uses more than one programming language,
such as analyzing data in one language but visualizing results in another,
make as clear of a division between the two as possible and document how to execute both. Frequently swapping back and forth between languages is a reproducibility nightmare
if it is not written in a fully stable and self-executing way,
because manually switching execution environments is very burdensome on others.
```